src/cache/precomputed_quantities.jl

```diff
@@ -112,11 +112,22 @@ function precomputed_quantities(Y, atmos)
     precipitation_quantities =
         atmos.precip_model isa Microphysics1Moment ?
         (; ᶜwᵣ = similar(Y.c, FT), ᶜwₛ = similar(Y.c, FT)) : (;)
+    cloud_diagnostics_quantities =
+        atmos.cloud_diagnostics isa DiagnosticCovariance ?
+        (;
+            ᶜqₜ′qₜ′ = similar(Y.c, FT),
+            ᶜθ′θ′ = similar(Y.c, FT),
+            ᶜθ′qₜ′ = similar(Y.c, FT),
+            cloud_fraction = similar(Y.c, FT)
+            ᶜqₜ_grad = (Y.f, Geometry.WVector{FT}),
+            ᶜθ_grad = (Y.f, Geometry.WVector{FT}),
+        ) : (;)
```

Comment:
We probably only need `cloud_fraction`. It seems we can use temporary fields for all other variables, or just inline them in the `get_cloud_fraction` function.